Mod page

matthias-luger opened the pull request:
fixes #881

matthias-luger committed:
mod page implementation

matthias-luger committed:
add rat check

matthias-luger committed:
added mod github link

matthias-luger committed:
added checking result elements

matthias-luger commented:
TODO: 
 - RAT checking API returns CORS error. Could/should be fixed in production?
 - More up-to-date youtube video / tutorial

Ekwav commented:
> @Ekwav The page is finished (UI-Side) but these things are still open:
> 
> * RAT checking API returns CORS error. => Could/should be fixed in production?
> * More up-to-date youtube video / tutorial => Do you plan on making one, or should we still use Tentamens'

* Talked to vytal, the api should work correctly now
* The installation from Tentamens is still up to date and will not change unless forge updates their website. Additionally you can link https://www.youtube.com/watch?v=dv66Y_cra8U&list=PLDpPmxIcq9tAssQlyJMBlSmSg5JOpq699

matthias-luger committed:
- add youtube tutorial playlist
- add mod page to navbar
- fix bug that breaks upload element after first upload
- added navbar to mod page

Ekwav commented on components/RatChecker/RatChecker.tsx:
```diff
@@ -0,0 +1,118 @@
+            )
+        }
+        if (checkingResult.rat.includes('Yes')) {
+            return <p style={{ color: 'red', fontSize: 'larger' }}>This mod file seems to be a rat. Be careful!</p>
```
```suggestion
            return <p style={{ color: 'red', fontSize: 'larger' }}>This mod is a known rat. We recommend against using it!</p>
```

Ekwav commented on components/RatChecker/RatChecker.tsx:
```diff
@@ -0,0 +1,118 @@
+            return <p style={{ color: 'red', fontSize: 'larger' }}>This mod file seems to be a rat. Be careful!</p>
+        }
+        if (checkingResult.rat.includes('No')) {
+            return <p style={{ color: 'lime', fontSize: 'larger' }}>This mod file doesn't seem to be a rat.</p>
```
```suggestion
            return <p style={{ color: 'lime', fontSize: 'larger' }}>No harmful code was found in this mod. It should be safe to use.</p>
```

Ekwav commented on components/ModDetails/ModDetails.tsx:
```diff
@@ -0,0 +1,53 @@
+                    <Card.Title>Details</Card.Title>
+                </Card.Header>
+                <Card.Body>
+                    <p>We provide a mod as a quality of life feature to use the flipper in minecraft directly.</p>
+                    <p>To use the mod you will need to link it to the website and verify your Minecraft account.</p>
```
The mod does way more than just provide flips in game. Also you do NOT need to link and/or verify.
```suggestion
                    <p>Say goodbye to window switching. The cofl mod provides you with data directly in Hypixel Skyblock. You can access price paid, the ah flipper, median, lbin, bazaar prices and more.</p>
                    <h3>Installation TLDR
                    <ol>
                    <li>Get and install Forge 1.8.9 (if you don't have already watch the video below)</li>
                    <li>Download and put the mod into your mods folder</li>
                    <li>Launch Minecraft and join Skyblock</li>
                    <li>Get a list of commands with /cofl help or watch the <a href="https://www.youtube.com/watch?v=dv66Y_cra8U&list=PLDpPmxIcq9tAssQlyJMBlSmSg5JOpq699">YouTube playlist</a> explaining them</li>
                    </ol>
```

Ekwav commented on components/ModDetails/ModDetails.tsx:
```diff
@@ -0,0 +1,53 @@
+                    <p>We provide a mod as a quality of life feature to use the flipper in minecraft directly.</p>
+                    <p>To use the mod you will need to link it to the website and verify your Minecraft account.</p>
+                    <hr />
+                    <p>
+                        The only legit place to download is on our{' '}
+                        <a target="_blank" rel="noreferrer" href="https://discord.gg/wvKXfTgCfb">
+                            Discord
+                        </a>{' '}
+                        in the #mod-releases channel.
+                    </p>
+                    <p>
+                        The entire mod is open source and you can check/compile it yourself here:{' '}
+                        <a target="_blank" rel="noreferrer" href="https://github.com/Coflnet/SkyblockMod">
+                            https://github.com/Coflnet/SkyblockMod
+                        </a>
+                    </p>
```
```suggestion
                    <p>
                         Mod downloads including source code can be found on the <a target="_blank" rel="noreferrer" href="https://github.com/coflnet/skyblockmod/releases  ">
                            Cofl Mod Github release page
                        </a>
                    </p>
```

Ekwav commented on components/RatChecker/RatChecker.tsx:
```diff
@@ -0,0 +1,118 @@
+                    <Card.Title>Rat Checker</Card.Title>
+                </Card.Header>
+                <Card.Body>
+                    <p>
+                        Please only download the mod from our{' '}
+                        <a target="_blank" rel="noreferrer" href="https://discord.gg/wvKXfTgCfb">
+                            Discord
+                        </a>
+                        . The download link is in the #mod-releases channel.
+                    </p>
```
```suggestion
```
This is for general rat checking, a link to our mod should only be a sideeffect not the first line.  
Also only download from the github https://github.com/coflnet/skyblockmod/releases

Ekwav commented on components/RatChecker/RatChecker.tsx:
```diff
@@ -0,0 +1,118 @@
+                    </p>
+                    {!isChecking ? (
+                        <div>
+                            <p>If you still want to check if a mod file is a RAT (remote access trojan), you can upload the file here:</p>
```
```suggestion
                            <p>Received a suspicious (or any other mod) that you want to check? Select it here and get quick results for known rats or legitimate mods (you can select multiple at once):</p>
```

Ekwav commented on components/RatChecker/RatChecker.tsx:
```diff
@@ -0,0 +1,118 @@
+                            {getResultElement()}
+                            <p style={{ float: 'right' }}>
+                                <small>
+                                    Files are checked by{' '}
```
```suggestion
                                    File check is provided by{' '}
```

Ekwav commented on components/RatChecker/RatChecker.tsx:
```diff
@@ -0,0 +1,118 @@
+                            <Form.Control type="file" className={'form-control'} ref={ratFileInput} onChange={onFileUpload} />
+                        </div>
+                    ) : (
+                        getLoadingElement(<p>Checking file</p>)
```
please allow checking multiple files at once

Ekwav commented on components/ModDetails/ModDetails.tsx:
```diff
@@ -0,0 +1,53 @@
+                            allowFullScreen
+                        ></iframe>
+                    </div>
+                    <p style={{ marginTop: '50px' }}>
+                        We also have a playlist on how to use different ingame commands:
+                        <br />
+                        <a target="_blank" rel="noreferrer" href="https://www.youtube.com/watch?v=dv66Y_cra8U&list=PLDpPmxIcq9tAssQlyJMBlSmSg5JOpq699">
+                            {'https://www.youtube.com/watch?v=dv66Y_cra8U&list=PLDpPmxIcq9tAssQlyJMBlSmSg5JOpq699'}
+                        </a>
+                    </p>
```
```suggestion
```

Ekwav commented on components/ModDetails/ModDetails.tsx:
```diff
@@ -0,0 +1,53 @@
+                    </p>
+                    <hr />
+                    <div style={{ display: 'flex', justifyContent: 'center' }}>
+                        <iframe
+                            width="560"
+                            height="315"
+                            src="https://www.youtube.com/embed/qwnvobFTcS0"
+                            title="YouTube video player"
+                            frameBorder="0"
+                            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+                            allowFullScreen
+                        ></iframe>
```
can this be made bigger when clicked? Like scaling it to full width of the container

matthias-luger replied:
Why? There is the option to full-screen it.
Also: I can't detect if you play the video, just if you somewhere clicked on it.
I would not expect a embeded player to change size when I click on it.

Ekwav commented on components/ModDetails/ModDetails.tsx:
```diff
@@ -0,0 +1,53 @@
+        <>
+            <Card>
+                <Card.Header>
+                    <Card.Title>Details</Card.Title>
```
```suggestion
                    <Card.Title>Cofl skyblock mod details</Card.Title>
```

Ekwav commented on pages/mod.tsx:
```diff
@@ -0,0 +1,25 @@
+            <Container>
+                <h2>
+                    <NavBar />
+                    Mod
```
```suggestion
                    CoflMod
```

matthias-luger committed:
Update components/RatChecker/RatChecker.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
Update components/RatChecker/RatChecker.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
Update components/ModDetails/ModDetails.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
Update components/ModDetails/ModDetails.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
Update components/RatChecker/RatChecker.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
Update components/ModDetails/ModDetails.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
allow multi file upload

matthias-luger committed:
Merge branch 'mod-page' of https://github.com/Coflnet/hypixel-react into mod-page

matthias-luger committed:
fix html errors
remove TLDR

matthias-luger committed:
Update components/ModDetails/ModDetails.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
Update pages/mod.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

Ekwav commented on components/NavBar/NavBar.tsx:
```diff
@@ -183,6 +184,9 @@ function NavBar(props: Props) {
                         <MenuItem icon={<PetsIcon />}>
                             <Link href={'/kat'}>Kat flips</Link>
                         </MenuItem>
+                        <MenuItem icon={<FileIcon />}>
```
please use a download icon

matthias-luger committed:
change file icon to download icon

matthias-luger committed:
Update components/RatChecker/RatChecker.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
Update components/RatChecker/RatChecker.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>
